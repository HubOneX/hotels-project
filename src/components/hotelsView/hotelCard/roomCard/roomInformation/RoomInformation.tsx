import { Typography } from "@mui/material";
import React, { FC } from "react";
import breakpoints from "../../../../../constants/breakpoints";
import { Occupancy } from "../../../../../types/DataTypes";
import { FlexColumnContainer } from "../../../../utilityComponents/styledComponents";
import styled from "@emotion/styled";

const RoomInfoContainer = styled(FlexColumnContainer)`
  @media (max-width: ${breakpoints.md}px) {
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;
    gap: 20px;
    padding: 0;
  }
`;

type Props = {
  name: string;
  occupancy: Occupancy;
};

const RoomInformation: FC<Props> = ({ name, occupancy }) => {
  return (
    <RoomInfoContainer>
      <Typography
        variant="subtitle1"
        sx={{ marginRight: "auto", width: { xs: "50%", sm: "auto" } }}
      >
        {name}
      </Typography>

      <Typography variant="subtitle2">Adults: {occupancy.maxAdults}</Typography>

      <Typography variant="subtitle2">
        Children: {occupancy.maxChildren}
      </Typography>
    </RoomInfoContainer>
  );
};

export default RoomInformation;
