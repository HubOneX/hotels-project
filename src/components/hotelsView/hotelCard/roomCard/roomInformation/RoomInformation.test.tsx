import RoomInformation from "./RoomInformation";
import renderer from "react-test-renderer";

describe("RoomInformation", () => {
  const defaultProps = {
    name: "Room",
    occupancy: {
      maxAdults: 2,
      maxChildren: 1,
      maxOverall: 3,
    },
  };
  it("should match the snapshot", () => {
    const container = renderer.create(<RoomInformation {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });
});
