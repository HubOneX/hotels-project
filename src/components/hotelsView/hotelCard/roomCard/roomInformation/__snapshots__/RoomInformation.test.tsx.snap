// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomInformation should match the snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-pwxgjm-MuiContainer-root"
>
  <h6
    className="MuiTypography-root MuiTypography-subtitle1 css-1f9ga5n-MuiTypography-root"
  >
    Room
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
  >
    Adults: 
    2
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
  >
    Children: 
    1
  </h6>
</div>
`;
